functions/src/auto-importer.ts

```diff
@@ -285,7 +309,7 @@ export const syncToS3AfterSyncDocWritten = functions.firestore
                     .catch(functions.logger.error)
                 } else {
                   // if the learner has no answers associated with this run, delete the doc
-                  deleteFromS3(context.params.runKey, data.resource_url)
+                  deleteFromS3(data.answer_metadata)
```

Comment:
Should this be returned? And possibly `syncToS3` above? I'm wondering how onWrite handles promises that haven't resolved yet.

Reply:
Hmmm, we do the same with the call to `syncToS3` in the line just above this (307) and it appears to work fine (both in the successful case and the case where the write has some error).

The `onWrite` is always getting some return value, either the `getAllAnswersForLearner...` promise or `null`.

So the question is whether `getAllAnswersForLearner...then` should return the promise inside `then`. It's possible that you're right that it should, although fwiw Typescript doesn't actually like me returning those promises:

```
Argument of type '(querySnapshot: QuerySnapshot<DocumentData>) => Promise<void | WriteResult> | Promise<void | DeleteObjectCommandOutput>' is not assignable to parameter of type '(value: QuerySnapshot<DocumentData>) => void | WriteResult | PromiseLike<void | WriteResult>'.
```

This might be easily fixable by typing the promises somehow.

Reply:
Yeah I guess we should just leave it as is since it is working. 🤷 
My best guess is that if there was a problem we'd find some cases where did_sync was not set because firestore killed off the function before it finished writing to S3.  So if we find those kind of issues we might want to revisit this.